proposals/README.md

```diff
@@ -0,0 +1,146 @@
+- reserve GitHub issues for tracking work in flight, instead of creating "umbrella"
+  issues
+- ensure community participants can successfully drive changes to
+  completion across one or more releases while stakeholders are adequately
+  represented throughout the process
+
+This process is supported by a unit of work of the Kubernetes Enhancement Proposal format, or KEP for short.
+A KEP attempts to combine aspects of
+
+- a feature, and effort-tracking document
+- a product requirements document
+- a design document
+
+into one file.
+
+We are proposing usage of KEPs in Kubeflow to be better aligne with the wider Kubernetes community.
+
+This proposal takes heavy inspiration from the [Kubernetes KEP Proposal](https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/0000-kep-process/README.md)
+
+## Motivation
+
+Documenting proposals and enhancements are critical in managing the complexity and longevity of software projects.
+
+Some examples include:
+
+1. Capturing the "why" behind critical architectural choices, not just the what.
+   1. This helps current and future team members understand the reasoning behind decisions, particularly when the rationale is no longer obvious.
+2. Improve Communication and Collaboration.
+3. Serve as a single source of truth for architectural decisions.
+4. By documenting options and their trade-offs, KEPs encourage structured decision-making and transparency.
+5. Enable Traceability
+6. Create a decision history that allows new (and old) contributors to trace architectural choices back to their original context, assumptions, and goals
+
+### Goals
+
+- Capture the Why Behind Decisions
+- Foster Clear Communication
+- Enable Decision Traceability
+- Encourage Thoughtful, Deliberate Decisions
+- Preserve Institutional Knowledge
+
+### Non-Goals
+
+- Not a substitute for technical or user documentation
+- Not a substitute or replacement for meaningful commit messages
+
+## Proposal
+
+A KEP is broken into sections which can be merged into source control
+incrementally in order to support an iterative development process. The KEP process
+is intended to create high-quality, uniform design and implementation documents
+for maintainers to deliberate. Contributors proposing new KEPs should create a copy of the [template directory](./NNNN-template)
+and proceed as per the instructions in the template.
+
+## Reference-Level Explanation
+
+### What Type of Work Should Be Tracked by a KEP
+
+Roughly any user or operator facing enhancement should follow the KEP process.
+If an enhancement would be described in either written or verbal communication
+to anyone besides the KEP author or developer, then consider creating a KEP.
+
+Similarly, any technical effort (refactoring, major architectural change) that
+will impact a large section of the development community should also be
+communicated widely. The KEP process is suited for this even if it will have
+zero impact on the typical user or operator.
```

Comment:
In practice i think this will not always happen if there is not enough bandwidth. So i think this should be for massive changes that are not straightforward.

"Similarly, technical effort (refactoring, major architectural change) that
will **significantly** impact a large section of the development community should also be
communicated widely. The KEP process is suited for this even if it will have
zero impact on the typical user or operator."

Reply:
That's actually fair. I don't want to make it significantly harder for community members to contribute to Kubeflow. I'll make the suggested edit.